classes/task/metadata_extraction_task.php

```diff
@@ -131,7 +131,7 @@ public function execute() {
                     $extraction->set('reason', get_string('status:nometadata', 'tool_metadata',
                         ['resourceid' => $data->resourceid, 'type' => $data->type]));
                 }
-            } catch (\moodle_exception $ex) {
+            } catch (\moodle_exception | \Throwable $ex) {
```

Comment:
```suggestion
            } catch (\Throwable $ex) {
```
This is to support all php versions.

Reply:
Ahh yes, indeed this was introduced in [7.1.0 release](https://www.php.net/ChangeLog-7.php#PHP_7_1): Implemented the RFC \`Catching multiple exception types\`.

Amending.